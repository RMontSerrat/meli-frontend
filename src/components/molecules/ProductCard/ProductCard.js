import React from 'react';
import PropTypes from 'prop-types';
import imgShipping from '../../../assets/images/ic_shipping.png';
import styles from './ProductCard.module.scss';
import { Price } from '~/components/atoms';

const ProductCard = ({
  imgUrl, price, cityName, name, freeShipping, onClick,
}) => (
  <div
    className={styles.ProductCard}
    data-testid="ProductCard"
    onClick={onClick}
    role="button"
    onKeyDown={onClick}
    tabIndex={0}
  >
    <img src={imgUrl} alt={name} title={name} className={styles.imgProduct} />
    <div className={styles.informations}>
      <div className={styles.priceContainer}>
        <Price {...price} />
        {freeShipping && (
          <figure className={styles.freeShipping} data-testid="FreeShipping">
            <img src={imgShipping} alt="Mercado Livre - Frete GrÃ¡tis" className={styles.imgShipping} />
          </figure>
        )}
      </div>
      <h2>{name}</h2>
    </div>
    <span className={styles.location}>
      {cityName}
    </span>
  </div>
);

ProductCard.propTypes = {
  imgUrl: PropTypes.string.isRequired,
  cityName: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  onClick: PropTypes.func,
  freeShipping: PropTypes.bool,
  price: PropTypes.shape({
    currency: PropTypes.string.isRequired,
    amount: PropTypes.number.isRequired,
    decimals: PropTypes.number.isRequired,
    variant: PropTypes.oneOf(['default', 'big']),
  }).isRequired,
};

ProductCard.defaultProps = {
  freeShipping: false,
  onClick: () => {},
};

export default ProductCard;
