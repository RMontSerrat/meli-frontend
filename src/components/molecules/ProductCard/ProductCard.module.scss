.ProductCard {
  cursor: pointer;
  display: grid;
  grid-template-columns: auto 3fr 1fr;
  outline: none;
  width: 100%;

  .imgProduct {
    height: rem(160px);
    object-fit: contain;
    width: rem(160px);
  }

  .informations {
    padding-left: rem(10px);

    h2 {
      font-family: $font-family;
      font-size: size(big);
      font-weight: $font-regular;
      margin: 0;
      padding-top: rem(10px);
    }
  }

  .priceContainer {
    align-items: flex-end;
    display: flex;
    padding-top: rem(10px);
  }

  .location {
    font-family: $font-family;
    font-weight: $font-thin;
    padding-top: rem(20px);
  }

  .freeShipping {
    margin: 0;
    padding-left: rem(10px);
  }

  .imgShipping {
    height: rem(20px);
    width: rem(20px);
  }

  @media (max-width: breakpoint(medium)) {
    grid-template-areas: 'a b'
      '. d';
    grid-template-columns: auto 2fr;
    grid-template-rows: repeat(2, auto);

    img {
      grid-area: a;
    }

    .informations {
      grid-area: b;
    }

    .location {
      grid-area: d;
      padding: 0 0 0 rem(10px);
    }
  }
  @media (max-width: breakpoint(mini)) {
    img {
      height: rme(80px);
      width: rem(80px);
    }
  }
}
