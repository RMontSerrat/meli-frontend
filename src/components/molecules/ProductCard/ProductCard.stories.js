/* eslint-disable */
import React from 'react';
import { storiesOf } from '@storybook/react';
import ProductCard from './ProductCard';

storiesOf('ProductCard', module).add('default', () => (
  <ProductCard
    imgUrl="https://http2.mlstatic.com/D_NQ_NP_600943-MLB31871248861_082019-V.webp" 
    price={{
      currency: 'R$',
      amount: 1000,
      decimals: 12,
    }}
    cityName="Rio de Janeiro"
    freeShipping
    name="Bateria Notebook - Dell Inspiron 14 3442 (14.8v) - Preta"
  />
));
