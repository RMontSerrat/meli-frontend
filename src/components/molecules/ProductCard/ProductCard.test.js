import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ProductCard from './ProductCard';

const product = {
  imgUrl: 'https://http2.mlstatic.com/D_NQ_NP_600943-MLB31871248861_082019-V.webp',
  price: {
    currency: 'R$',
    amount: 1000,
    decimals: 12,
  },
  cityName: 'Rio de Janeiro',
  name: 'Bateria Notebook - Dell Inspiron 14 3442 (14.8v) - Preta',
};

describe('<ProductCard />', () => {
  test('it should mount', () => {
    render(<ProductCard {...product} />);

    const productCard = screen.getByTestId('ProductCard');

    expect(productCard).toBeInTheDocument();
  });

  test('snapshot', () => {
    const { asFragment } = render(
      <ProductCard {...product} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('snapshot with freeShipping', () => {
    const { asFragment } = render(
      <ProductCard {...product} freeShipping />,
    );

    const freeShipping = screen.getByTestId('FreeShipping');

    expect(freeShipping).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
