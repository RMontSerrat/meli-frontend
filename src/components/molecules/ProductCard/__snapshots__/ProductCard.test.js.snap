// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductCard /> snapshot 1`] = `
<DocumentFragment>
  <div
    class="ProductCard"
    data-testid="ProductCard"
    role="button"
    tabindex="0"
  >
    <figure
      class="ProgressiveImage imgProduct"
      data-testid="ProgressiveImage"
    >
      <span
        class="overlayImage"
      />
      <img
        alt="Bateria Notebook - Dell Inspiron 14 3442 (14.8v) - Preta"
        src="https://http2.mlstatic.com/D_NQ_NP_600943-MLB31871248861_082019-V.webp"
        title="Bateria Notebook - Dell Inspiron 14 3442 (14.8v) - Preta"
      />
    </figure>
    <div
      class="informations"
    >
      <div
        class="priceContainer"
      >
        <span
          class="Price default"
          data-testid="Price"
        >
          <span
            class="currency"
          >
            R$
          </span>
          <span
            class="amount"
          >
            1.000
          </span>
          <span
            class="decimals"
          >
            12
          </span>
        </span>
      </div>
      <h2>
        Bateria Notebook - Dell Inspiron 14 3442 (14.8v) - Preta
      </h2>
    </div>
    <span
      class="location"
    >
      Rio de Janeiro
    </span>
  </div>
</DocumentFragment>
`;

exports[`<ProductCard /> snapshot with freeShipping 1`] = `
<DocumentFragment>
  <div
    class="ProductCard"
    data-testid="ProductCard"
    role="button"
    tabindex="0"
  >
    <figure
      class="ProgressiveImage imgProduct"
      data-testid="ProgressiveImage"
    >
      <span
        class="overlayImage"
      />
      <img
        alt="Bateria Notebook - Dell Inspiron 14 3442 (14.8v) - Preta"
        src="https://http2.mlstatic.com/D_NQ_NP_600943-MLB31871248861_082019-V.webp"
        title="Bateria Notebook - Dell Inspiron 14 3442 (14.8v) - Preta"
      />
    </figure>
    <div
      class="informations"
    >
      <div
        class="priceContainer"
      >
        <span
          class="Price default"
          data-testid="Price"
        >
          <span
            class="currency"
          >
            R$
          </span>
          <span
            class="amount"
          >
            1.000
          </span>
          <span
            class="decimals"
          >
            12
          </span>
        </span>
        <figure
          class="freeShipping"
          data-testid="FreeShipping"
        >
          <img
            alt="Mercado Livre - Frete GrÃ¡tis"
            class="imgShipping"
            src="ic_shipping.png"
          />
        </figure>
      </div>
      <h2>
        Bateria Notebook - Dell Inspiron 14 3442 (14.8v) - Preta
      </h2>
    </div>
    <span
      class="location"
    >
      Rio de Janeiro
    </span>
  </div>
</DocumentFragment>
`;
